Q-11.

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats

Reading the data

In [4]:
df=pd.read_csv(r"C:\Users\XMALL\Downloads\data.csv")

In [5]:
df.head()

Patient ID   Blood Pressure Before (mmHg)   Blood Pressure After (mmHg)
0           1                            130                           120
1           2                            142                           135
2           3                            120                           118
3           4                            135                           127
4           5                            148                           140

***Check whether the distribution of all the classes are the same or not.***



There are no classes in the given data. However to check the distribution of numeric variables accross multiple classes on a target variable, we use ANOVA (Analysis of Variance) test. Which uses the f-distribution curve to ascertain weather the mean values of different classes are statistically similar or different. ANOVA test checks the squared dispersion within the same class and among two separate classes.

***Check for the equality of variance***

In [10]:
statistic, p_value = stats.levene(df[' Blood Pressure Before (mmHg)'], df[' Blood Pressure After (mmHg)'])
statistic, p_value

(0.18038002140150966, 0.6715080090945376)

We see that the p-value (0.6715) is greater than the commonly taken significance level of 5%. Therefore, the variances are equal in the before and the after data.


***Which amount LDA and QDA would perform better on this data for classification and why.***

We cannot perform LDA (Linear discriminant analysis) or QDA (Quardratic discriminant analysis) for classification of the given dataset as these methods are for classification of categorical data, but the given data set has only numeric features, which is not suitable to apply on LDA or QDA.

***Check the equality of mean for between all the classes.***

We don't have categorical features in our dataset so we cannot perform such a check. However, such a check can be performed by a one-way ANOVA test.